### PR TITLE
fix(backup): Better sorting for sorted_dependencies()

### PR DESCRIPTION
**Pull request by azaslavsky**
Previously, the sorting between sibling dependencies (that is, dependencies which could be swapped in the final sorted list without affecting he validity of the topographic sort) was a bit haphazard, and could be thrown of by faraway changes like import order. We now do more alphabetization when sorting, so this is no longer an issue.

The user-facing message that tests raise when the dependency graph has been changed has also been improved.

### CODE REVIEW

File: src/sentry/backup/dependencies.py [hunk 1]
```diff
@@ -512,7 +512,9 @@ def sorted_dependencies() -> list[Type[models.base.Model]]:
     Similar to Django's algorithm except that we discard the importance of natural keys
     when sorting dependencies (ie, it works without them)."""
 
-    model_dependencies_dict = list(dependencies().values())
+    model_dependencies_dict = list(
```
Comment (asottile-sentry) [inline src/sentry/backup/dependencies.py:515]: this variable seems incorrectly named -- it isn't a dict

`list(sorted(...))` is redundant -- `sorted` returns a list

Comment (asottile-sentry) [reply src/sentry/backup/dependencies.py]: applied my feedback as a patch

**Review by asottile-sentry — APPROVED** [review]

File: src/sentry/backup/dependencies.py (hunk 1 above)
Comment (azaslavsky) [reply src/sentry/backup/dependencies.py]: Thanks!

**Pull request merged**